embedding/csharp/Tizen.Flutter.Embedding/ElmFlutterView.cs

```diff
@@ -0,0 +1,193 @@
+
+using System;
+using System.Diagnostics;
+using ElmSharp;
+using static Tizen.Flutter.Embedding.Interop;
+
+namespace Tizen.Flutter.Embedding
+{
+    /// <summary>
+    /// Represents an <see cref="EvasObject"/> instance created by the embedder.
+    /// </summary>
+    class EvasObjectImpl : EvasObject
+    {
+        public EvasObjectImpl(EvasObject parent, IntPtr handle) : base(parent)
+        {
+            Handle = handle;
+        }
+
+        protected override IntPtr CreateHandle(EvasObject parent)
+        {
+            return Handle;
+        }
+    }
+
+    /// <summary>
+    /// Displays a Flutter screen in a Tizen application.
+    /// </summary>
+    public class ElmFlutterView : IPluginRegistry
+    {
+        /// <summary>
+        /// The Flutter engine instance.
+        /// </summary>
+        private FlutterEngine Engine { get; set; } = null;
+
+        /// <summary>
+        /// The Flutter view instance handle.
+        /// </summary>
+        private FlutterDesktopView View { get; set; } = new FlutterDesktopView();
+
+        /// <summary>
+        /// The backing Evas object for this view.
+        /// </summary>
+        public EvasObject EvasObject { get; private set; } = null;
+
+        /// <summary>
+        /// The parent of <see cref="EvasObject"/>.
+        /// </summary>
+        private EvasObject Parent { get; set; } = null;
+
+        /// <summary>
+        /// The initial width of the view. Defaults to the parent width if the value is zero.
+        /// </summary>
+        private int InitialWidth { get; set; } = 0;
+
+        /// <summary>
+        /// The initial height of the view. Defaults to the parent height if the value is zero.
+        /// </summary>
+        private int InitialHeight { get; set; } = 0;
+
+        /// <summary>
+        /// Whether the view is running.
+        /// </summary>
+        public bool IsRunning => !View.IsInvalid;
+
+        /// <summary>
+        /// The current width of the view.
+        /// </summary>
+        public int Width
+        {
+            get
+            {
+                Debug.Assert(IsRunning);
+
+                return EvasObject.Geometry.Width;
+            }
+        }
+
+        /// <summary>
+        /// The current height of the view.
+        /// </summary>
+        public int Height
+        {
+            get
+            {
+                Debug.Assert(IsRunning);
+
+                return EvasObject.Geometry.Height;
+            }
+        }
+
+        public ElmFlutterView(EvasObject parent)
+        {
+            Parent = parent;
+        }
+
+        public ElmFlutterView(EvasObject parent, int initialWidth, int initialHeight)
+        {
+            Parent = parent;
+            InitialWidth = initialWidth;
+            InitialHeight = initialHeight;
+        }
+
+        /// <summary>
+        /// Starts running the view with the associated engine, creating if not set.
+        /// </summary>
+        public bool RunEngine()
```

Comment:
Should we prevent calling RunEngine() twice?
`IsRunning` can be used for it.

Reply:
Okay. We may need some unit tests as well.